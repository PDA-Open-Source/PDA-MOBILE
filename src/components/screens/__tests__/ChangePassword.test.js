import React from 'react';
import renderer from 'react-test-renderer';
import ChangePassword from '../ChangePassword';
import {shallow } from 'enzyme';
jest.mock('react-native-device-info');
jest.mock('../../../utils/StaticFunctions');

describe('SignUp Page Component',()=>{
  let wrapper ;
  beforeEach(()=>{
    wrapper = shallow(<ChangePassword/>);
  });
  describe('SignUp Page Component rendering',()=>{

    it('Check ChangePassword render without break', () =>{
      const tree = renderer.create(<ChangePassword/>).toJSON();
      expect(tree).toBeTruthy();
    });

    it('Change Password Component should render ',()=>{
      const snap = renderer.create(<ChangePassword/>);
      expect(snap.toJSON()).toMatchSnapshot();
    });

    it('render 3 CustomTextInput',()=>{
      expect(wrapper.find('CustomTextInput').length).toBe(3);
    });

    it('render one Button ',()=>{
      expect(wrapper.find('Button').length).toBe(1);
    });


  });

});