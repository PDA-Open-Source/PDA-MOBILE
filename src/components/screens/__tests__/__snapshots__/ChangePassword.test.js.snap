// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Page Component SignUp Page Component rednering Change Password Component should render  1`] = `
<View
  pointerEvents={null}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#607D8B",
        "elevation": 4,
        "flexDirection": "row",
        "paddingTop": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 0.5,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "height": 56,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "textAlign": "center",
              },
              Object {
                "fontFamily": "Roboto-Medium",
                "fontSize": 31.428571428571427,
              },
            ]
          }
        >
          CHANGE PASSWORD
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 25,
              },
              Array [
                undefined,
              ],
              Object {
                "fontFamily": "Font Awesome 5 Pro",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "400",
              },
            ]
          }
        >
          ïƒ³
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 39.23529411764706,
        "paddingVertical": 62.77647058823529,
      }
    }
  >
    <Text
      style={
        Array [
          undefined,
          Object {
            "fontFamily": "Roboto-Regular",
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "color": "#26A69A",
            "marginBottom": 39.23529411764706,
            "textAlign": "center",
          },
        ]
      }
    >
      After Successful Change Password, you will redirect to login page and you need to login again.
    </Text>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#fff",
            "flexDirection": "row",
            "marginVertical": 19.61764705882353,
          },
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#888888",
            "flexDirection": "row",
            "marginVertical": 19.61764705882353,
          },
        ]
      }
    >
      <TextInput
        accessibilityLabel="currentPassword_Input"
        allowFontScaling={true}
        autoCapitalize="none"
        keyboardType="default"
        maxLength={250}
        multiline={false}
        numberOfLines={1}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Current Password"
        placeholderTextColor="#AAAAAA"
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#fff",
              "flex": 1,
            },
            Object {
              "fontFamily": "Roboto-Regular",
              "fontSize": 23.57142857142857,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontFamily": "Roboto-Light",
            },
            Object {
              "color": "#000",
              "paddingHorizontal": 39.23529411764706,
            },
          ]
        }
        testID="currentPassword_Input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#fff",
            "flexDirection": "row",
            "marginVertical": 19.61764705882353,
          },
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#888888",
            "flexDirection": "row",
            "marginVertical": 19.61764705882353,
          },
        ]
      }
    >
      <TextInput
        accessibilityLabel="currentPassword_Input"
        allowFontScaling={true}
        autoCapitalize="none"
        keyboardType="default"
        maxLength={250}
        multiline={false}
        numberOfLines={1}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="New Password"
        placeholderTextColor="#AAAAAA"
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#fff",
              "flex": 1,
            },
            Object {
              "fontFamily": "Roboto-Regular",
              "fontSize": 23.57142857142857,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontFamily": "Roboto-Light",
            },
            Object {
              "color": "#000",
              "paddingHorizontal": 39.23529411764706,
            },
          ]
        }
        testID="password_Input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#fff",
            "flexDirection": "row",
            "marginVertical": 19.61764705882353,
          },
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#888888",
            "flexDirection": "row",
            "marginVertical": 19.61764705882353,
          },
        ]
      }
    >
      <TextInput
        accessibilityLabel="retypePassword_Input"
        allowFontScaling={true}
        autoCapitalize="none"
        keyboardType="default"
        maxLength={250}
        multiline={false}
        numberOfLines={1}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Retype Password"
        placeholderTextColor="#AAAAAA"
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#fff",
              "flex": 1,
            },
            Object {
              "fontFamily": "Roboto-Regular",
              "fontSize": 23.57142857142857,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontFamily": "Roboto-Light",
            },
            Object {
              "color": "#000",
              "paddingHorizontal": 39.23529411764706,
            },
          ]
        }
        testID="retypePassword_Input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginTop": 15,
          "paddingBottom": 78.47058823529412,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      >
        <View>
          <View
            accessible={true}
            onLayout={[Function]}
            onPress={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="rgba(0, 0, 0,0.5)"
            rippleDuration={400}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={0}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#888888",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 5,
                "paddingHorizontal": 49.04411764705882,
                "paddingVertical": 10,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#888888",
                      "fontSize": 25,
                    },
                    Object {},
                    Object {
                      "fontFamily": "Font Awesome 5 Pro",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {
                      "fontWeight": "400",
                    },
                  ]
                }
              >
                ï…·
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <View
          accessible={true}
          onLayout={[Function]}
          onPress={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onRippleAnimation={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-only"
          rippleCentered={false}
          rippleColor="#546E7A"
          rippleDuration={400}
          rippleFades={true}
          rippleOpacity={0.3}
          rippleSequential={false}
          rippleSize={0}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#009688",
              "borderRadius": 40,
              "paddingHorizontal": 2,
              "paddingVertical": 15,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "color": "#fff",
                    "fontWeight": "bold",
                    "paddingHorizontal": 78.47058823529412,
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 28.285714285714285,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ],
                Object {
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 23.57142857142857,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            SAVE
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "borderRadius": 78.47058823529412,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
